### PR TITLE
Run linter

### PR DESCRIPTION
**Pull request by synthtech**
Fixed most errors caught by `npm run lint`:
- [no-extra-semi](https://eslint.org/docs/rules/no-extra-semi)
- [no-useless-escape](https://eslint.org/docs/rules/no-useless-escape)
- [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins)
- [no-redeclare](https://eslint.org/docs/rules/no-redeclare)
- [no-mixed-spaces-and-tabs](https://eslint.org/docs/rules/no-mixed-spaces-and-tabs)
- [no-cond-assign](https://eslint.org/docs/rules/no-cond-assign)
- [no-empty](https://eslint.org/docs/rules/no-empty)
- [no-inner-declarations](https://eslint.org/docs/rules/no-inner-declarations)

The majority of it was semicolon stuff.

Before: 50 errors, 462 warnings
After: 6 errors, 0 warnings

Of the errors that remain:
- it doesn't like `return` used in `if` statements at the top level in https://github.com/hohMiyazawa/Automail/blob/156b8f1e362a1d44a8705edecad7d8ab8ba1d678/src/settings.js
- a few things in https://github.com/hohMiyazawa/Automail/blob/156b8f1e362a1d44a8705edecad7d8ab8ba1d678/src/utilities/levDist.js and https://github.com/hohMiyazawa/Automail/blob/156b8f1e362a1d44a8705edecad7d8ab8ba1d678/src/utilities/lz-string.js

The above utilities seem to be third-party code so I was unsure if I should add them to the ignore file or not.

### CODE REVIEW

**Comment by hohMiyazawa** [comment]
Most of these are good to go right away, but things like changing regexes has potential to introduce subtle changes, so it could take some days before I have reviewed all of it.

File: src/polyfills.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ if(!String.prototype.includes){//https://developer.mozilla.org/en-US/docs/Web/Ja
 if(!window.BroadcastChannel){
     var channels = [];
 
-    function BroadcastChannel(channel) {
+    const BroadcastChannel = function(channel) {
         var $this = this;
```
Comment (hohMiyazawa) [inline src/polyfills.js:20]: TODO: check if this affects the value of ``this``

File: src/modules/settingsPage.js [hunk 2]
```diff
@@ -671,7 +671,7 @@ query{
 					alert("error parsing JSON")
 					return
 				}
-				if(!data.hasOwnProperty("automailAPI")){//sanity check
+				if(!Object.prototype.hasOwnProperty.call(data, "automailAPI")){//sanity check
```
Comment (hohMiyazawa) [inline src/modules/settingsPage.js:674]: I see this construction used several places. What does it do?

**Comment by hohMiyazawa** [comment]
Is there any good source for javscript regex escape rules?

**Comment by hohMiyazawa** [comment]
Top level returns are acceptable, since the script is wrapped in a function.

**Comment by hohMiyazawa** [comment]
Would probably be more clear to have a helper method for this hard to grasp ``Object.prototype.hasOwnProperty.call()`` things. Most of the time, what's needed would most clearly be expressed by ``hasProperty(thing,"property")``, with ``hasProperty`` implemented in the utilities somewhere.
In some cases, were falsy values do not cause problems, they can also be replaced by a simple ``if(thing.property)`` (but that's hard to say if unfamiliar with the allowed values).

**Comment by synthtech** [comment]
>Is there any good source for javscript regex escape rules?

I'm not sure, from a quick search:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
https://learnbyexample.github.io/learn_js_regexp/escaping-metacharacters.html

>Add levDist and lz-string to ignored third party code.

will do 👍 

>Top level returns are acceptable, since the script is wrapped in a function.

I see, will probably have to add an inline comment to ignore it.

>Would probably be more clear to have a helper method for this hard to grasp `Object.prototype.hasOwnProperty.call()` things. 

There's actually a solution to this already implemented in most current browsers: `Object.hasOwn()`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn
https://caniuse.com/mdn-javascript_builtins_object_hasown
It could be used with this polyfill: https://github.com/tc39/proposal-accessible-object-hasownproperty/blob/c02a5cab48b3fdeeed3180b68c5d0630c4d46269/polyfill.js

**Comment by hohMiyazawa** [comment]
``hasOwn`` would be the best name. No strong feelings wheter to polyfill it, or use the polyfill code to make a ``hasOwn`` utility.

**Comment by hohMiyazawa** [comment]
I think I have checked all the regexes now.

File: src/polyfills.js (hunk 1 above)
Comment (hohMiyazawa) [reply src/polyfills.js:20]: seems fine

File: src/modules/settingsPage.js (hunk 2 above)
Comment (hohMiyazawa) [reply src/modules/settingsPage.js:674]: see later hawOwn discussion

**Comment by synthtech** [comment]
>or use the polyfill code to make a hasOwn utility.

I think that's doable. Could be something like:

```js
function hasOwn(obj, prop){
	if(!Object.hasOwn){
		return Object.prototype.hasOwnProperty.call(obj, prop)
	}
	else{
		return Object.hasOwn(obj, prop)
	}
}
```

**Comment by hohMiyazawa** [comment]
Yes, exactly what I had in mind.

**Comment by hohMiyazawa** [comment]
Why are the queries marked as ignored? They aren't third party code.
Is it just their structure that makes it hard to make the linter work properly?

**Comment by synthtech** [comment]
> Is it just their structure that makes it hard to make the linter work properly?

Yup, they all get parser errors because it can't understand the surrounding object syntax (which isn't a problem in the final build)

**Comment by hohMiyazawa** [comment]
Then everything seems in order. Merged.

**Pull request merged**